src/sentry/snuba/metrics/query.py

```diff
@@ -139,7 +140,13 @@ def validate_orderby(self) -> None:
             metric_mri = get_mri(f.field.metric_name)
             # Construct a metrics expression
             metric_field_obj = metric_object_factory(f.field.op, metric_mri)
-            entity = metric_field_obj.get_entity(self.project_ids)
+
+            # Find correct use_case_id based on metric_name
+            use_case_id = UseCaseKey.RELEASE_HEALTH
+            if f.field.metric_name.startswith("transaction"):
```

Comment:
There is no guarantee that the public name won't change. The only guarantee exists on MRI not changing, and so please check against name space in MRI regex not on the metric name

Reply:
fixed.